contracts/BC_fusion/StakeHub.sol

```diff
@@ -315,6 +324,24 @@ contract StakeHub is System, Initializable, Protectable {
     }
 
     /*----------------- external functions -----------------*/
+    function updateAgent(address newAgent) external validatorExist(msg.sender) {
+        if (agentToOperator[newAgent] != address(0)) revert InvalidAgent();
+        if (_validatorSet.contains(newAgent)) revert InvalidAgent();
+
+        address operatorAddress = msg.sender;
+        address oldAgent = _validators[operatorAddress].agent;
+        if (oldAgent == newAgent) revert InvalidAgent();
+
+        if (oldAgent != address(0)) {
+            agentToOperator[oldAgent] = address(0);
```

Comment:
why not use `delete agentToOperator[oldAgent]`

Reply:
> why not use `delete agentToOperator[oldAgent]`

agree

Reply:
fixed